crates/sui-indexer-alt-graphql/src/api/types/balance_change.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::Object;
+use sui_indexer_alt_schema::transactions::BalanceChange as NativeBalanceChange;
+use sui_types::object::Owner as NativeOwner;
+
+use crate::{api::scalars::big_int::BigInt, scope::Scope};
+
+use super::address::Address;
+
+#[derive(Clone)]
+pub(crate) struct BalanceChange {
+    pub(crate) scope: Scope,
+    pub(crate) native: NativeBalanceChange,
+}
+
+/// Effects to the balance (sum of coin values per coin type) of addresses and objects.
+#[Object]
+impl BalanceChange {
+    /// The address or object whose balance has changed.
+    async fn owner(&self) -> Option<Address> {
+        use NativeOwner as O;
+        let NativeBalanceChange::V1 { owner, .. } = &self.native;
+
+        match owner {
+            O::AddressOwner(addr)
+            | O::ObjectOwner(addr)
+            | O::ConsensusAddressOwner { owner: addr, .. } => {
+                Some(Address::with_address(self.scope.clone(), *addr))
+            }
+            O::Shared { .. } | O::Immutable => None,
+        }
+    }
+
+    /// The inner type of the coin whose balance has changed (e.g. `0x2::sui::SUI`).
+    async fn coin_type(&self) -> Option<String> {
```

Comment:
This should be a `MoveType` -- let's leave it out for now and put a TODO here instead so we don't forget to add it when we have support for types.